Utils/azure-toolkit-ide-hdinsight-libs/azure-toolkit-ide-cosmos-spark-lib/src/main/java/com/microsoft/azure/toolkit/ide/cosmos/spark/SparkOnCosmosActionsContributor.java

```diff
@@ -0,0 +1,173 @@
+import com.microsoft.azure.toolkit.lib.common.action.Action;
+import com.microsoft.azure.toolkit.lib.common.action.ActionGroup;
+import com.microsoft.azure.toolkit.lib.common.action.AzureActionManager;
+import com.microsoft.azure.toolkit.lib.common.action.IActionGroup;
+import com.microsoft.azure.toolkit.lib.common.model.AbstractAzResourceModule;
+import com.microsoft.azure.toolkit.lib.common.model.AzResource;
+import com.microsoft.azure.toolkit.lib.resource.ResourceGroup;
+import com.microsoft.azure.toolkit.lib.sparkoncosmos.SparkOnCosmosADLAccountNode;
+import com.microsoft.azure.toolkit.lib.sparkoncosmos.SparkOnCosmosClusterModule;
+import com.microsoft.azure.toolkit.lib.sparkoncosmos.SparkOnCosmosClusterNode;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+
+import java.awt.*;
+import java.io.IOException;
+import java.net.URI;
+
+public class SparkOnCosmosActionsContributor implements IActionsContributor {
+    private static final String SPARK_NOTEBOOK_LINK = "https://aka.ms/spkadlnb";
+    public static final int INITIALIZE_ORDER = ResourceCommonActionsContributor.INITIALIZE_ORDER + 1;
+    public static final String SERVICE_ACTIONS = "actions.sparkoncosmos.service";
+    public static final Action.Id<ResourceGroup> GROUP_CREATE_SOC_SERVICE = Action.Id.of("user/sparkoncosmos.create_sparkoncosmos.group");
+    public static final String ADLA_NODE_ACTIONS = "actions.sparkoncosmos.adla";
+    public static final String CLUSTER_NODE_ACTIONS = "actions.sparkoncosmos.cluster";
+    public static final Action.Id<Object> PROVISION_CLUSTER = Action.Id.of("user/sparkoncosmos.provision_cluster.spark");
+    public static final Action.Id<Object> SUBMIT_SOC_SERVERLESS_JOB = Action.Id.of("user/sparkoncosmos.submit_serverlessjob.spark");
+    public static final Action.Id<Object> VIEW_SOC_SERVERLESS_JOB = Action.Id.of("user/sparkoncosmos.view_serverlessjob.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> DELETE_CLUSTER = Action.Id.of("user/sparkoncosmos.delete_cluster.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> OPEN_SPARK_HISTORY_UI = Action.Id.of("user/sparkoncosmos.open_history_ui.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> OPEN_SPARK_MASTER_UI = Action.Id.of("user/sparkoncosmos.open_master_ui.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> VIEW_CLUSTER_STATUS = Action.Id.of("user/sparkoncosmos.view_cluster_status.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> UPDATE_CLUSTER = Action.Id.of("user/sparkoncosmos.update_cluster.spark");
+    public static final Action.Id<SparkOnCosmosClusterNode> SUBMIT_CLUSTER_JOB = Action.Id.of("user/sparkoncosmos.submit_cluster_job.spark");
+
+
+    public static final Action.Id<Object> OPEN_NOTEBOOK = Action.Id.of("user/sparkoncosmos.open_notebook.spark");
+
+    @Override
+    public void registerActions(AzureActionManager am) {
+        new Action<>(OPEN_NOTEBOOK)
```

Comment:
you have to config id parameter with `withIdParam(...)` because the action id has parameters.